### PR TITLE
xmrmaker: check if swap was completed already when restarting swap

### PR DESCRIPTION
**Pull request by noot**
as title says, I came across an issue when testing where the maker claimed ETH, but then shutdown/didn't get the result and mark the swap complete due to a bad endpoint. Then on restart, it would be waiting for a ready/refunded event, but neither of these would ever happen b/c the swap was already done.

this PR updates the xmrmaker to check if the swap was already completed and the ETH was claimed already (by them) on restart, and if so marks the swap successfully complete.

### CODE REVIEW

File: protocol/xmrmaker/errors.go
Comment (dimalinux) [inline protocol/xmrmaker/errors.go:33]: I'm in favor of standardized naming since we have several ID values:
OfferID (or Offer ID)
SwapID (or Swap ID)
PeerID (or Peer ID)

In this case, you can extract that it is the SwapID, because we don't have Offer and Peer IDs in the contract, but with standardized naming, the reader has to think less.

File: protocol/xmrmaker/swap_state.go
Comment (dimalinux) [inline protocol/xmrmaker/swap_state.go:170]: Can this 90 line function be broken up at all? The name says "check", implying that is is read-only, but if you scroll down enough, it is also updating the state of the swap.

Comment (noot) [reply protocol/xmrmaker/swap_state.go]: yeah, I broke it into two, one that just checks if the swap is already completed (funds claimed) and one that updates the state

File: protocol/xmrmaker/errors.go
Comment (noot) [reply protocol/xmrmaker/errors.go:33]: you mean naming the contract swap ID `SwapID`? I'd rather name it something more explicit like `OnChainID` or `ContractSwapID` (this is usually used in places anyways)

I think it's a bit out of scope for this PR but we can open an issue for it!

Comment (dimalinux) [reply protocol/xmrmaker/errors.go:33]: No, I was proposing that we be more rigorous in the order of the words so readers don't have to carefully parse sentences and so they can search the code base for keywords and word sequences. While I do like the phrase `OnChainID`, the code base is currently using the phrase "Swap ID" and sometimes "Contract Swap ID" to refer to the on-chain ID.  The error message can be rewritten from `swap with given ID in contract does not exist` to `contract swap ID does not exist`.

File: protocol/xmrmaker/instance_test.go
Comment (dimalinux) [inline protocol/xmrmaker/instance_test.go:111]: Not related to this PR, but should we have changed `common.DefaultEthEndpoint` to use the `http` endpoint after the recent pull request removing the websockets dependency?

File: protocol/xmrmaker/instance_test.go
Comment (dimalinux) [inline protocol/xmrmaker/instance_test.go:227]: These might be more helpful at the test package level. Your call. I've been putting them in `init` methods at the test package level.  I'm slightly tempted to put the function that sets them all in `common`, and then have `main` and the tests use the same function to enable logging levels.

File: protocol/xmrmaker/swap_state.go
Comment (dimalinux) [inline protocol/xmrmaker/swap_state.go:216]: This is declared as `*ethtypes.Log`, but the code is only using it as a boolean.  You could declare `claimFound := false` here and just set it to `true` when you find it, and then you won't need the nil check at the end.

**Review by dimalinux — APPROVED** [review]
Approved, but make sure to look at the comments and maybe take action on some of them before merging.

File: protocol/xmrmaker/swap_state.go
Comment (dimalinux) [inline protocol/xmrmaker/swap_state.go:245]: The note explains "what it does", but I suspect that there is a reason "why" which might also be helpful for the reader.

File: protocol/xmrmaker/errors.go
Comment (noot) [reply protocol/xmrmaker/errors.go:33]: ah, i see. okay i'll update the message!

File: protocol/xmrmaker/instance_test.go
Comment (noot) [reply protocol/xmrmaker/instance_test.go:111]: yeah we can, either one should still work though

File: protocol/xmrmaker/instance_test.go
Comment (noot) [reply protocol/xmrmaker/instance_test.go:227]: ah I just put these here to debug, will remove

**Pull request merged**
